crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -1084,11 +965,37 @@ impl actor::HcP2p for HolochainP2pActor {
 
     fn get(
         &self,
-        _dna_hash: DnaHash,
-        _dht_hash: holo_hash::AnyDhtHash,
-        _options: actor::GetOptions,
+        dna_hash: DnaHash,
+        dht_hash: holo_hash::AnyDhtHash,
+        options: actor::GetOptions,
     ) -> BoxFut<'_, HolochainP2pResult<Vec<WireOps>>> {
-        Box::pin(async move { todo!() })
+        Box::pin(async move {
+            let kitsune = self.kitsune()?;
+            let space_id = dna_hash.to_k2_space();
+            let space = kitsune.space(space_id.clone()).await?;
+            let loc = dht_hash.get_loc();
+
+            let (to_agent, to_url) = self.get_peer_for_loc("get", &space, loc).await?;
+
+            let r_options: event::GetOptions = (&options).into();
+
+            let (msg_id, req) = crate::wire::WireMessage::get_req(to_agent, dht_hash, r_options);
+
+            let start = std::time::Instant::now();
+
+            let out = self
+                .send_request("get", &space, to_url, msg_id, req, |res| match res {
+                    crate::wire::WireMessage::GetRes { response, .. } => Ok(vec![response]),
+                    _ => Err(HolochainP2pError::other(format!(
+                        "invalid response to get: {res:?}"
+                    ))),
+                })
+                .await;
```

Comment:
I think this is essentially what current holochain (like v 0.4) is doing. That is, just returning an error from the first peer it is able to make the get request to.

But we should consider if this is what we want it to be doing. Because we could loop here for the duration of the timeout trying additional peers or something.

edit: (deleted the previous version of this comment that was on the wrong line and moved it here : )

Reply:
Yes, that sounds right. All the "try again and hope it picks a different peer" logic lives in Holochains workflows for now